### PR TITLE
fix(wsgi): WSGI integrations respect SCRIPT_NAME env variable

### PR DESCRIPTION
**Pull request by sarvaSanjay**
URLs generated using Sentry's WSGI Middleware should include SCRIPT_NAME in the event's url

Fixes https://github.com/getsentry/sentry-python/issues/2576

Noticed that the problem described in the above issue wasn't specific to Django per se but general to all WSGI applications which is why I have made modifications to the SentryWsgiMiddleware class that should resolve this issue for Django and other integrations as well. As far as I can tell this problem should not arise for ASGI applications as the root_path is accounted for.

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

**Comment by sarvaSanjay** [comment]
Hey, its my first contribution. It seems as if we need to add a label to this PR for any of the checks to execute, could any of the collaborators add the same?

**Comment by szokeasaurusrex** [comment]
> Hey, its my first contribution. It seems as if we need to add a label to this PR for any of the checks to execute, could any of the collaborators add the same?

I have approved most of the tests to run. For now, the AWS permissions check will still fail – we will add the necessary label once we have reviewed and approved the change

File: sentry_sdk/integrations/wsgi.py [hunk 1]
```diff
@@ -62,7 +62,14 @@ def get_request_url(environ, use_x_forwarded_for=False):
...
         get_host(environ, use_x_forwarded_for),
-        wsgi_decoding_dance(environ.get("PATH_INFO") or "").lstrip("/"),
+        wsgi_decoding_dance(
+            "/".join(
+                [
+                    environ.get("SCRIPT_NAME", "").rstrip("/"),
+                    environ.get("PATH_INFO", "").lstrip("/"),
+                ]
+            )
+        ).lstrip("/"),
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/wsgi.py:72]: The following might be more readable. For readability, I would also suggest computing the path before the return statement and storing it in a variable (named `path` or something similar); then, the call could be replaced with `wsgi_decoding_dance(path).lstrip("/")`.
```suggestion
        wsgi_decoding_dance(
            "%s/%s" % (
                environ.get("SCRIPT_NAME", "").rstrip("/"),
                environ.get("PATH_INFO", "").lstrip("/"),
            )
        ).lstrip("/"),
```

File: tests/integrations/wsgi/test_wsgi.py [hunk 2]
```diff
@@ -67,6 +67,29 @@ def test_basic(sentry_init, crashing_app, capture_events):
...
+    events = capture_events()
+
+    with pytest.raises(ZeroDivisionError):
+        # setting url with PATH_INFO: bark/, HTTP_HOST: dogs.are.great and SCRIPT_NAME: woof/woof/
+        client.get("bark/", "https://dogs.are.great/woof/woof/")
+
+    (event,) = events
+
+    assert event["transaction"] == "generic WSGI request"
+    print(event["request"])
```
Comment (szokeasaurusrex) [inline tests/integrations/wsgi/test_wsgi.py:83]:
```suggestion
```
This `print` statement should probably be removed

File: tests/integrations/wsgi/test_wsgi.py [hunk 3]
```diff
@@ -67,6 +67,29 @@ def test_basic(sentry_init, crashing_app, capture_events):
...
+    sentry_init(send_default_pii=True)
+    app = SentryWsgiMiddleware(crashing_app)
+    client = Client(app)
+    events = capture_events()
+
+    with pytest.raises(ZeroDivisionError):
+        # setting url with PATH_INFO: bark/, HTTP_HOST: dogs.are.great and SCRIPT_NAME: woof/woof/
+        client.get("bark/", "https://dogs.are.great/woof/woof/")
+
+    (event,) = events
```
Comment (szokeasaurusrex) [inline tests/integrations/wsgi/test_wsgi.py:80]:
```suggestion
    assert len(events) == 1
    (event,) = events
```
Assuming we are expecting to have exactly one event, we should explicitly assert this expectation to document this expectation in case the test fails. Otherwise, if there isn't exactly one event, the `(event,) = event` line will error, and it might be unclear to whoever is debugging the error whether there is an error in the code or whether the test is implemented incorrectly.

File: tests/integrations/wsgi/test_wsgi.py [hunk 4]
```diff
@@ -67,6 +67,29 @@ def test_basic(sentry_init, crashing_app, capture_events):
...
+
+    assert event["transaction"] == "generic WSGI request"
+    print(event["request"])
+    assert event["request"] == {
+        "env": {"SERVER_NAME": "dogs.are.great", "SERVER_PORT": "443"},
+        "headers": {"Host": "dogs.are.great"},
+        "method": "GET",
+        "query_string": "",
+        "url": "https://dogs.are.great/woof/woof/bark/",
+    }
```
Comment (szokeasaurusrex) [inline tests/integrations/wsgi/test_wsgi.py:90]: Since the purpose of this test is to check that we set the URL correctly, I would suggest only checking that the URL is set correctly, like so:
```suggestion
    assert event["request"]["url"] == "https://dogs.are.great/woof/woof/bark/"
```

This way, we don't have to update the test in case we were to add/change the other fields on the event request dictionary.

File: tests/integrations/wsgi/test_wsgi.py [hunk 5]
```diff
@@ -67,6 +67,29 @@ def test_basic(sentry_init, crashing_app, capture_events):
     }
 
 
+def test_basic_script_name_is_respected(sentry_init, crashing_app, capture_events):
```
Comment (szokeasaurusrex) [inline tests/integrations/wsgi/test_wsgi.py:70]: Maybe we could add additional test cases for edge cases

**Review by szokeasaurusrex — COMMENTED** [review]
Thank you for the PR! It looks good, but I think we should make a couple of improvements before we merge. Please see my inline comments

**Comment by szokeasaurusrex** [comment]
Also, please address the linter and test failures (except the AWS Lambda tests – you can ignore these for now); we can only merge the changes once all these tests pass

**Review by szokeasaurusrex — APPROVED** [review]
